backend/pkg/dbschemas/sql/postgresql/queries/system.sql

```diff
@@ -680,3 +679,29 @@ GROUP BY
     referenced_schema.nspname,
     referenced_tbl.relname,
     ref_columns.foreign_column_names;
+
+
+-- name: GetUniqueIndexesBySchema :many
+SELECT
+  ns.nspname AS table_schema,                      -- Schema name for the table
+  tbl.relname AS table_name,                         -- Name of the table the index belongs to
+  idx.relname AS index_name,                         -- Name of the index
+  array_agg(col.attname ORDER BY key_info.ordinality)::TEXT[] AS index_columns  -- Comma-separated list of index columns
+FROM pg_catalog.pg_class AS tbl
+  -- Join to get the schema information for the table
+  JOIN pg_catalog.pg_namespace AS ns ON tbl.relnamespace = ns.oid
+  -- Join to retrieve index metadata for the table
+  JOIN pg_catalog.pg_index AS idx_meta ON tbl.oid = idx_meta.indrelid
+  -- Join to get the index object details
+  JOIN pg_catalog.pg_class AS idx ON idx_meta.indexrelid = idx.oid
+  -- Unnest the index key attribute numbers along with their ordinal positions
+  JOIN unnest(idx_meta.indkey) WITH ORDINALITY AS key_info(attnum, ordinality) ON true
+  -- Join to get the column attributes corresponding to the index keys
+  JOIN pg_catalog.pg_attribute AS col ON col.attrelid = tbl.oid AND col.attnum = key_info.attnum
+WHERE ns.nspname = ANY(sqlc.arg('schema')::TEXT[])
+  AND idx_meta.indisunique = true
+  -- Exclude indexes automatically created by constraints
+  AND NOT EXISTS (
+       SELECT 1 FROM pg_catalog.pg_constraint AS cons WHERE cons.conindid = idx.oid
```

Comment:
Hm, the thing that might get us here is we saw with $customer that they had a unique index that we were excluding from our index query because it was created by a constraint. However, that constraint no longer existed, but the index was around..and there were FK's that were utilizing this unique index for their references.

I'm wondering now if it would actually be valuable to include these indices here. 
That way, if the original constraint exists, great the algo will use that. If not, then it falls back on the dangling index. @alishakawaguchi

Reply:
yeah that makes sense 👍

Reply:
fixed